### DISCUSSION TITLE
Add support for transformRequest

### DISCUSSION THREAD

**Discussion by yardenhod** [rnmapbox/maps#1225] [Ideas]
transformRequest would be very helpful for passing request params, using credentials,...
It will be nice if this option will be available like in the original mapbox package for react:

```
// use transformRequest to modify requests that begin with `http://myHost`
transformRequest: function(url, resourceType) {
 if (resourceType === 'Source' && url.indexOf('http://myHost') > -1) {
   return {
     url: url.replace('http', 'https'),
     headers: { 'my-custom-header': true },
     credentials: 'include'  // Include cookies for cross-origin requests
   }
  }
 }
```

please consider to add this feature :)

**Comment by vamshi9666** [comment]
any work around for this ?

**Reply by mfazekas** [reply]
See https://github.com/react-native-mapbox-gl/maps/blob/master/docs/CustomHttpHeaders.md

**Reply by vamshi9666** [reply]
I am using aws location services , and going through [this](https://github.com/aws-samples/amazon-location-samples/blob/main/maplibre-gl-js-react/src/index.tsx) approach. And not able to fit customHttpHeaders to fit it.

**Comment by RGkevin** [comment]
Any update on this? I need this to production release. Thank you!
